pints/_mcmc/_emcee_hammer.py

```diff
@@ -101,6 +99,14 @@ def ask(self):
         # Return proposed points
         return self._proposed
 
+    def _sample_z(self, a):
+        """
+        Samples ``z~g(z)`` where ``g(z)`` is proportional to ``1 / sqrt(z)``
+        if ``z`` is in ``[1 / a, a]``; otherwise 0.
+        """
+        r = np.random.rand()
+        return ((1 + r * (a - 1))**2) / a
```

Comment:
Could you explain how this formula relates to g(z) propto 1/sqrt(z) and the interval condition on z?

Reply:
I've added a note to indicate that the method uses inverse transform sampling. I'm not sure I want to explain the formula itself though...